VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/ExplicitAccessModifiers/ExplicitAccessModifiersAnalyzer.cs

```diff
@@ -0,0 +1,221 @@
+                        accessibilityKeyword));
+                }
+            }
+
+            if (context.Node is EventFieldDeclarationSyntax)
+            {
+                var declarationExpression = (EventFieldDeclarationSyntax)context.Node;
+                if (!declarationExpression.Modifiers.Any(m => _modifierKinds.Contains(m.Kind())))
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Rule, declarationExpression.GetLocation(),
+                        "private"));
+                }
+            }
+
+            if (context.Node is EventDeclarationSyntax)
+            {
+                var declarationExpression = (EventDeclarationSyntax)context.Node;
+                if (!declarationExpression.Modifiers.Any(m => _modifierKinds.Contains(m.Kind())))
+                {
+                    var accessibility = context.SemanticModel.GetDeclaredSymbol(declarationExpression).DeclaredAccessibility;
+                    var accessibilityKeyword = AccessibilityToString(accessibility);
+
+                    context.ReportDiagnostic(Diagnostic.Create(Rule, declarationExpression.GetLocation(),
+                        accessibilityKeyword));
+                }
+            }
+
+            if (context.Node is IndexerDeclarationSyntax)
+            {
+                var declarationExpression = (IndexerDeclarationSyntax)context.Node;
+                if (!declarationExpression.Modifiers.Any(m => _modifierKinds.Contains(m.Kind())))
+                {
+                    var accessibility = context.SemanticModel.GetDeclaredSymbol(declarationExpression).DeclaredAccessibility;
+                    var accessibilityKeyword = AccessibilityToString(accessibility);
+
+                    context.ReportDiagnostic(Diagnostic.Create(Rule, declarationExpression.GetLocation(),
+                        accessibilityKeyword));
+                }
+            }
+        }
+
+        private readonly SyntaxKind[] _modifierKinds =
+        {
+            SyntaxKind.PublicKeyword,
+            SyntaxKind.ProtectedKeyword,
+            SyntaxKind.InternalKeyword,
+            SyntaxKind.PrivateKeyword
+        };
+
+        private string AccessibilityToString(Accessibility accessibility)
+        {
+            switch (accessibility)
+            {
+                case Accessibility.Private:
+                    return "private";
+                case Accessibility.ProtectedAndInternal:
+                    return "protected internal";
+                case Accessibility.Protected:
+                    return "protected";
+                case Accessibility.Internal:
+                    return "internal";
+                case Accessibility.Public:
+                    return "public";
+                default:
+                    // friend has the same value as internal, and not applicable isn't an option here
+                    return "something happened - please create an issue on our GitHub page";
```

Comment:
Throw an exception instead, this would generate a general error anyway since it wouldn't be able to parse the tree. Something like `throw new ArgumentException(nameof(accessibility));`